common/src/io.cpp

```diff
@@ -182,7 +182,7 @@ pcl::concatenateFields (const pcl::PCLPointCloud2 &cloud1,
  
   // Iterate over each point and perform the appropriate memcpys
   int point_offset = 0;
-  for (std::size_t cp = 0; cp < cloud_out.width * cloud_out.height; ++cp)
+  for (std::size_t cp = 0; cp < static_cast<uindex_t>(cloud_out.width * cloud_out.height); ++cp)
```

Comment:
Type of `cp` should be `uindex_t`? (Conclusion reached here needs to be applied to lots of places)

Reply:
You mean change type of `cp` to `uindex_t`? Wouldn't changing it to `index_t` be more consistent with what we've been doing elsewhere? Or do you think we should use type `uindex_t` in all such scenarios (since `cp` here should never be negative)?

Reply:
`cp` refers to memory location, not index, which for `index_t` points can spill over to `uindex_t`. Doesn't matter for 64 bit, but matters for 32 bit

Reply:
Are you sure? From the following idiom `cp * point_step` I would  say the interpretation is that it is an index. Is your statement based on it not being used inside an `operator[]` for pcl types.

Reply:
It's based on the class being `PCLPointCLoud2` which is a binary format

Reply:
I'm back. Based on the code, it seems `cp` is a counter for points and thus, should be `index_t`